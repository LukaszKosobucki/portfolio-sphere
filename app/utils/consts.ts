export type Socials = {
  NAME: string;
  ICON: string;
  TEXT: string;
  HREF: string;
}[];
export type Page = {
  TITLE: string;
  DESCRIPTION: string;
};

export interface Site extends Page {
  AUTHOR: string;
}

export const WORK: Page = {
  TITLE: "Work",
  DESCRIPTION: "Places I have worked.",
};

// Projects Page
export const PROJECTS: Page = {
  TITLE: "Projects",
  DESCRIPTION: "Recent projects I have worked on.",
};
