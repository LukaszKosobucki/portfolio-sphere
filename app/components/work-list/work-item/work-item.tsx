import { formatDate } from "@/app/utils/format-date";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
import { Document, INLINES } from "@contentful/rich-text-types";
import { Entry, EntrySkeletonType } from "contentful";

function WorkItem({
  entry,
}: {
  entry: Entry<EntrySkeletonType, undefined, string>;
}) {
  const renderOptions = {
    renderNode: {
      [INLINES.HYPERLINK]: (node: any, children: any) => (
        <a href={node.data.uri} target="_blank" rel="noopener noreferrer">
          {children}
        </a>
      ),
    },
  };

  return (
    <li className="animate border-b border-black/10 dark:border-white/25 mt-4 first-of-type:mt-0 first-of-type:pt-0 last-of-type:border-none">
      <div className="text-sm uppercase mb-4">
        {formatDate(entry.fields.dateStart as string)} -{" "}
        {entry.fields.dateEnd ? formatDate(entry.fields.dateEnd as string) : "present"}
      </div>
      <div className="text-black dark:text-white font-semibold">{entry.fields.name as string}</div>
      <div className="text-sm font-semibold">{entry.fields.workTitle as string}</div>
      <article className="prose dark:prose-invert prose-p:text-justify">
        {documentToReactComponents(entry.fields.description as Document, renderOptions)}
      </article>
    </li>
  );
}

export default WorkItem;
