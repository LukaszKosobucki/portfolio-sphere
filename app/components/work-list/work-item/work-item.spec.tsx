import { formatDate } from "@/app/utils/format-date";
import { formatSnapshot } from "@/app/utils/tests/format-snapshot";
import { mockEntryWithoutEndDate, mockWorkItem } from "@/app/utils/tests/mocks/work-item.mock";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
import { render, screen } from "@testing-library/react";
import { afterAll, beforeEach, describe, expect, jest, mock, test } from "bun:test";
import WorkItem from "./work-item";

mock.module("@/app/utils/format-date", () => ({
  formatDate: jest.fn(),
}));

mock.module("@contentful/rich-text-react-renderer", () => ({
  documentToReactComponents: jest.fn(),
}));

describe("WorkItem Component", () => {
  beforeEach(() => {
    (formatDate as jest.Mock).mockImplementation((date: string) => date);
    (documentToReactComponents as jest.Mock).mockImplementation(() => <div>Mock Description</div>);
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(await WorkItem({ entry: mockWorkItem }));

    expect(formatSnapshot(asFragment())).toMatchSnapshot();
  });

  test("should render formatted start and end dates", () => {
    render(<WorkItem entry={mockWorkItem} />);

    expect(formatDate).toHaveBeenCalledWith("2023-01-04");
    expect(formatDate).toHaveBeenCalledWith("2023-08-04");

    expect(screen.getByText("2023-01-04 - 2023-08-04")).toBeInTheDocument();
  });

  test('should render "continuing" if dateEnd is not provided', () => {
    render(<WorkItem entry={mockEntryWithoutEndDate} />);

    expect(formatDate).toHaveBeenCalledWith("2023-09-04");
    expect(screen.getByText("2023-09-04 - present")).toBeInTheDocument();
  });

  test("should render the name and workTitle", () => {
    render(<WorkItem entry={mockWorkItem} />);

    expect(screen.getByText("YouGov")).toBeInTheDocument();
    expect(screen.getByText("Junior Software Engineer")).toBeInTheDocument();
  });

  test("should render the description", () => {
    render(<WorkItem entry={mockWorkItem} />);

    expect(documentToReactComponents).toHaveBeenCalledWith(
      mockWorkItem.fields.description,
      expect.any(Object),
    );
    expect(screen.getByText("Mock Description")).toBeInTheDocument();
  });
});
