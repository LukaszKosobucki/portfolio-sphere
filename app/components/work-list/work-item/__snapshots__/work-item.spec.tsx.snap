// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkItem Component should match snapshot 1`] = `
"<DocumentFragment>
  <li
    class="animate border-b border-black/10 dark:border-white/25 mt-4 first-of-type:mt-0 first-of-type:pt-0 last-of-type:border-none"
  >
    <div
      class="text-sm uppercase mb-4"
    >
      2023-01-04 - 2023-08-04
    </div>
    <div
      class="text-black dark:text-white font-semibold"
    >
      YouGov
    </div>
    <div
      class="text-sm font-semibold"
    >
      Junior Software Engineer
    </div>
    <article
      class="prose dark:prose-invert prose-p:text-justify"
    >
      <div>
        Mock Description
      </div>
    </article>
  </li>
</DocumentFragment>"
`;
